### ISSUE TITLE
Magento 2.3.2 (or higher) requires libsodium 1.0.13

### ISSUE THREAD

**Issue by shochdoerfer** [markshust/docker-magento#193]
Magento 2.3.2 (or higher) requires a newer version of libsodium as it is shipped with Debian Stretch. Version 1.0.11 which is shipped with Debian Stretch is lacking the constant `SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13` which Magento is looking for. Upgrading to Debian Buster should fix this as Buster ships with libsodium 1.0.17.

**Comment by markshust** [comment]
Thanks for this. I'm going to look into Buster again, I had a few issues with it in the past but we eventually need to upgrade it.

**Comment by markshust** [comment]
I don't believe Magento 2.3.2+ _requires_ the `SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13 ` constant:

```
    /**
     * Gets latest hash algorithm version.
     *
     * @return int
     */
    public function getLatestHashVersion(): int
    {
        if (extension_loaded('sodium') && defined('SODIUM_CRYPTO_PWHASH_ALG_ARGON2ID13')) {
            return self::HASH_VERSION_ARGON2ID13;
        }

        return self::HASH_VERSION_SHA256;
    }
```

You can see that it does fallback to `HASH_VERSION_SHA256`. This isn't ideal though, so I'm going to leave this ticket open until it is resolved.

I started working on some new images a while back which contain a lot of updates, but my focus was on the training material. I'll try to loop back to these updated images very soon as I definitely want this and many other updates released soon :)

**Comment by shochdoerfer** [comment]
Correct, it's not a strict requirement. I think my issue was that the database dump I used was done on a 7.3 installation with the sodium extension. That seems to lead to a side effect when changing a users password e.g. via magerun.

Need help getting the updated images out? Awesome training material btw :)

**Comment by markshust** [comment]
Yep, that will definitely do it. I'm adding a task label to this ticket. Thank you :)

I'm hoping to work on this in the next release. I'll try to keep this a minor update, remember having some issues with Buster before though.

**Comment by markshust** [comment]
All fixed with the new `27.0.0` docker compose setup https://github.com/markshust/docker-magento/blob/master/CHANGELOG.md#2700---2020-01-01

**Comment by shochdoerfer** [comment]
@markshust awsome! Thanks for taking care of this.
